src/common/components/clusterConfiguration/DiscoveryImageConfigForm.tsx

```diff
@@ -135,9 +131,16 @@ export const DiscoveryImageConfigForm: React.FC<DiscoveryImageConfigFormProps> =
               </Stack>
             </ModalBoxBody>
             <ModalBoxFooter>
-              <Button key="submit" onClick={submitForm}>
-                {t('ai:Generate Discovery ISO')}
-              </Button>
+              {!isSubmitting && (
+                <Button key="submit" onClick={submitForm}>
+                  {t('ai:Generate Discovery ISO')}
+                </Button>
+              )}
+              {isSubmitting && (
+                <Button key="generating" isLoading isDisabled>
+                  {t('ai: Generating')}
+                </Button>
+              )}
```

Comment:
`key` is unnecessary and we can make the code a bit more readable with:
```suggestion
              <Button onClick={submitForm} isDisabled={isSubmitting} isLoading={isSubmitting}>
               {isSubmitting ? t('ai: Generating') : t('ai:Generate Discovery ISO')}
             </Button>
```